opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -0,0 +1,411 @@
+        let mut blobs = Vec::new();
+        for ((schema_id, batch_event_name), (schema_entry, records)) in batches {
+            let events: Vec<CentralEventEntry> = records
+                .into_iter()
+                .map(|(event_name, level, row_buffer)| CentralEventEntry {
+                    schema_id,
+                    level,
+                    event_name,
+                    row: BondEncodedRow::from_schema_and_row(&schema_entry.schema, &row_buffer),
+                })
+                .collect();
+            let events_len = events.len();
+
+            let blob = CentralBlob {
+                version: 1,
+                format: 2,
+                metadata: metadata.to_string(),
+                schemas: vec![schema_entry],
+                events,
+            };
+            let bytes = blob.to_bytes();
+            blobs.push((schema_id, batch_event_name, bytes, events_len));
+        }
+        blobs
+    }
+
+    /// Determine which fields are present in the LogRecord
+    fn determine_fields(&self, log: &LogRecord) -> Vec<(String, u8)> {
+        let mut fields = vec![
+            ("env_name".to_string(), BondDataType::BT_STRING as u8),
+            ("env_ver".to_string(), BondDataType::BT_STRING as u8),
+            ("timestamp".to_string(), BondDataType::BT_STRING as u8),
+            ("env_time".to_string(), BondDataType::BT_STRING as u8),
+        ];
+
+        // Part A extension - Conditional fields
+        if !log.trace_id.is_empty() {
+            fields.push(("env_dt_traceId".to_string(), BondDataType::BT_STRING as u8));
+        }
+        if !log.span_id.is_empty() {
+            fields.push(("env_dt_spanId".to_string(), BondDataType::BT_STRING as u8));
+        }
+        if log.flags != 0 {
+            fields.push((
+                "env_dt_traceFlags".to_string(),
+                BondDataType::BT_INT32 as u8,
+            ));
+        }
+
+        // Part B - Core log fields
+        if !log.event_name.is_empty() {
+            fields.push(("name".to_string(), BondDataType::BT_STRING as u8));
+        }
+        fields.push(("SeverityNumber".to_string(), BondDataType::BT_INT32 as u8));
+        if !log.severity_text.is_empty() {
+            fields.push(("SeverityText".to_string(), BondDataType::BT_STRING as u8));
+        }
+        if let Some(body) = &log.body {
+            if let Some(Value::StringValue(_)) = &body.value {
+                fields.push(("body".to_string(), BondDataType::BT_STRING as u8));
+            }
+        }
+
+        // Part C - Dynamic attributes
+        for attr in &log.attributes {
+            if let Some(val) = attr.value.as_ref().and_then(|v| v.value.as_ref()) {
+                let type_id = match val {
+                    Value::StringValue(_) => BondDataType::BT_STRING as u8,
+                    Value::IntValue(_) => BondDataType::BT_INT32 as u8,
+                    Value::DoubleValue(_) => BondDataType::BT_FLOAT as u8, // using float for now
```

Comment:
Why not use `double`?

```suggestion
                    Value::DoubleValue(_) => BondDataType::BT_DOUBLE as u8, // using float for now
```